### PR TITLE
DEX: Remove `price` argument

### PR DESCRIPTION
**Pull request by Warchant**

### CODE REVIEW

**Review by uchu — APPROVED** [review]
Looks good

**Review by baidang201 — APPROVED** [review]

File: pallet/dex/src/lib.rs [hunk 1]
```diff
@@ -660,19 +659,24 @@ pub mod pallet {
...
+			let (a, b) = match order_type {
+				OrderType::SELL => (requested_amount, offered_amount),
+				OrderType::BUY => (offered_amount, requested_amount),
+			};
+
+			// because price is an integer, we need to check if the division is exact
+			// (does not have a remainder)
+			let price = a
+				.checked_div(&b)
+				.ok_or(Error::<T>::PriceDoNotMatchOfferedRequestedAmount)?;
```
Comment (akorchyn) [inline pallet/dex/src/lib.rs:671]: This error is division by zero, but up-to-you if you want to fix it.

For some reason, I thought rust had checked_div_mod. Not sure, where I have seen it...

**Review by akorchyn — APPROVED** [review]
Good job, small nitpick

File: pallet/dex/src/lib.rs (hunk 1 above)
Comment (Warchant) [reply pallet/dex/src/lib.rs:671]: image
(my screenshot tool went crazy, don't ask why the screenshot looks like this...)

**Pull request merged**